lib/addons/karpenter/index.ts

```diff
@@ -39,124 +65,203 @@ export interface EbsVolumeMapping {
  */
 export interface KarpenterAddOnProps extends HelmAddOnUserProps {
     /**
-     * Taints for the provisioned nodes - Taints may prevent pods from scheduling if they are not tolerated by the pod.
-     */
-    taints?: Taint[],
-
-    /**
-     * Provisioned nodes will have these taints, but pods do not need to tolerate these taints to be provisioned by this\
-     * provisioner. These taints are expected to be temporary and some other entity (e.g. a DaemonSet) is responsible for
-     * removing the taint after it has finished initializing the node.
-     */
-    startupTaints?: Taint[],
-
-    /**
-     * Labels applied to all nodes
-     */
-    labels?: Values,
-
-    /**
-     * Annotations applied to all nodes
-     */
-    annotations?: Values,
-
-    /**
-     * Requirement properties for a Provisioner (Optional) - If not provided, the add-on will
-     * deploy a Provisioner with default values.
-     */
-    requirements?: {
-        key: string,
-        op: 'In' | 'NotIn' ,
-        vals: string[],
-    }[]
-
-    /**
-     * Tags needed for subnets - Subnet tags and security group tags are required for the provisioner to be created
-     */
-    subnetTags?: Values,
-
-    /**
-     * Tags needed for security groups - Subnet tags and security group tags are required for the provisioner to be created
-     */
-    securityGroupTags?: Values,
-
-    /**
-     * AMI Family: If provided, Karpenter will automatically query the appropriate EKS optimized AMI via AWS Systems Manager
+     * This is the top level nodepool specification. Nodepools launch nodes in response to pods that are unschedulable. 
+     * A single nodepool is capable of managing a diverse set of nodes. 
+     * Node properties are determined from a combination of nodepool and pod scheduling constraints.
      */
-    amiFamily?: "AL2" | "Bottlerocket" | "Ubuntu" | "Windows2019" | "Windows2022"
+    NodePoolSpec?: {
```

Comment:
attributes lower case - `nodePoolSpec` just like the rest of the attributes. 

You can extract a type like 
`export type NodePoolSpec ...`, then use `nodePoolSpec: NodePoolSpec`.

The above approach with exported type allows configuring objects of type NodePoolSpec in customer projects that import blueprints. 

The way you declared it now - it creates an anonymous type that customers cannot use directly (not exported) unless they create an object of type `KarpenterAddOnProps`. 

Some customers brought it up, that if they want to reuse karpenter across several blueprints and just modify for example nodePoolSpec, it becomes a problem as they need to clone the whole outer structure.

Reply:
exported outside to SPI